src/core/reducers/search.js

```diff
@@ -1,34 +1,35 @@
 import {
   SEARCH_STARTED,
   SEARCH_LOADED,
-  SEARCH_FAILED,
 } from 'core/constants';
 
 const initialState = {
   count: 0,
   filters: {},
   loading: false,
-  page: 1,
   results: [],
 };
 
 export default function search(state = initialState, action) {
   const { payload } = action;
   switch (action.type) {
     case SEARCH_STARTED:
-      return { ...state, ...payload, loading: true };
+      return {
+        ...state,
+        count: payload.count,
+        filters: payload.filters,
+        results: [],
+        loading: true,
+      };
     case SEARCH_LOADED:
       return {
         ...state,
         count: payload.result.count,
         loading: false,
-        filters: payload.filters,
```

Comment:
I thought that too, but I didn't like dispatching the same data again when it wouldn't have changed (extra tests, code, for nothing really). If we change the search to allow multiple searches we'd not only need to change the saga but also our search code to hold multiple sets of results for different filters. So I figured it was better to remove it as they shouldn't change.

Writing a test to ensure filters haven't been overwritten is essentially writing our own test for saga's `takeLatest` which would be redundant and a bit messy/verbose, but for what it's worth in local testing the `takeLatest` saga definitely prevents old results overwriting new ones.